config.go

```diff
@@ -1362,10 +1355,26 @@ func extractBitcoindRPCParams(bitcoindConfigPath string) (string, string, string
 // ZMQ rawblock and rawtx notifications are different.
 func checkZMQOptions(zmqBlockHost, zmqTxHost string) error {
 	if zmqBlockHost == zmqTxHost {
-		return errors.New("zmqpubrawblock and zmqpubrawtx must be set" +
+		return errors.New("zmqpubrawblock and zmqpubrawtx must be set " +
 			"to different addresses")
 	}
 
+	zmqBlockHostURL, err := url.Parse(zmqBlockHost)
```

Comment:
Nit: could use `ResolveTCPAddr` instead to make a stricter check as that's what we require.

Reply:
@wpaulino you mean, to add `net.ResolveTCPAddr` in addition to `url.Parse`? The latter is needed anyway as the option value is in the URL format, e.g. `tcp://0.0.0.0:8334`

Reply:
I reworked the whole function, hope I checked all cases.

Reply:
Ah yes, that was my mistake, `net.ResolveTCPAddr` isn't needed here. The new changes should mostly be unnecessary now. `url.Parse` can't parse addresses without a scheme prefix, e.g. `0.0.0.0:8334`, so as long as that isn't `tcp`, we can just return an error like `invalid ZMQ address, must be TCP` or something like that.

Reply:
@wpaulino
> we can just return an error like `invalid ZMQ address, must be TCP`

Well, but why? Is it because LND currently can't use anything else than TCP?

Reply:
@wpaulino I found in tests that at least `ipc:///` is supported, so maybe we shouldn't be so strict about TCP-only sockets...

Reply:
Besides that, I believe `net.ResolveTCPAddr` is a good idea anyway, as it's better resolve once and then work with the IP instead of issuing DNS requests on every connection.